llama_index/node_parser/relational/unstructured_element.py

```diff
@@ -62,13 +62,18 @@ def html_to_df(html_str: str) -> pd.DataFrame:
         cols = [c.text.strip() if c.text is not None else "" for c in cols]
         data.append(cols)
 
+    """ Check if the  first row is as long as the data rows, if not likely not a table """
+    if len(data) > 0 and len(data[0]) != len(data[1]):
```

Comment:
understand that we want to address the issue of unstructured.

Reply:
Good point, but if all columns don't have the same length, it will crash panda / therefor llama as it will create a unvalid dataframe.

We could try to correct it by extending the number of columns is all rows, but then we may end up changing the meaning of the table as we have no way to guess which column is missing / merged in the html. This also lead to a lot of issue with table used for Formating (like in the Tesla 10Q notebook sample).

So in the end we end up not indexing the table that have inconsistent column length, but I add some code to cast them back as text nodes, so they will be in the document node collection, and available for retrieval, just not indexed as Table.

Reply:
thanks. Table is tricky. let us first fix that issue